src/librustc_trans/back/wasm.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fs;
+use std::path::Path;
+use std::collections::BTreeMap;
+
+use serialize::leb128;
+
+pub fn add_custom_sections(path: &Path, sections: &BTreeMap<String, Vec<u8>>) {
+    let mut wasm = fs::read(path).expect("failed to read wasm output");
+
+    // see https://github.com/WebAssembly/design/blob/master/BinaryEncoding.md#module-structure
```

Comment:
Can we really use https://webassembly.github.io/spec/core/binary/modules.html#custom-section for the reference?